lib/ip.rb

```diff
@@ -0,0 +1,14 @@
+module IP
+  require 'ipaddr'
+  class IP
+    # This constructor accepts both CIDR-notated IP and also exact IP.
+    def initialize(ip)
+      @address = IPAddr.new(ip)
+    end
+
+    def to_binary
+      binary = @address.to_i.to_s(2)
+      binary.slice(0, @address.prefix)
```

Comment:
Since `IPAddr#prefix` was introduced in Ruby 2.5.0, you have to provide an alternative way to acquire the prefix length of IP addresses. Unless `prefix` is defined, define the method on the class (monkey-patching).